render/detailed/summary.go

```diff
@@ -53,33 +64,58 @@ func MakeColumn(id string) Column {
 type NodeSummary struct {
 	ID           string        `json:"id"`
 	Label        string        `json:"label"`
-	Linkable     bool          `json:"linkable"` // Whether this node can be linked-to
+	LabelMinor   string        `json:"label_minor"`
+	Rank         string        `json:"rank"`
+	Shape        string        `json:"shape,omitempty"`
+	Stack        bool          `json:"stack,omitempty"`
+	Linkable     bool          `json:"linkable,omitempty"` // Whether this node can be linked-to
+	Pseudo       bool          `json:"pseudo,omitempty"`
 	Metadata     []MetadataRow `json:"metadata,omitempty"`
 	DockerLabels []MetadataRow `json:"docker_labels,omitempty"`
 	Metrics      []MetricRow   `json:"metrics,omitempty"`
+	Adjacency    report.IDList `json:"adjacency,omitempty"`
 }
 
 // MakeNodeSummary summarizes a node, if possible.
-func MakeNodeSummary(n render.RenderableNode) (NodeSummary, bool) {
-	renderers := map[string]func(report.Node) NodeSummary{
+func MakeNodeSummary(n report.Node) (NodeSummary, bool) {
+	renderers := map[string]func(NodeSummary, report.Node) (NodeSummary, bool){
+		render.Pseudo:         pseudoNodeSummary,
 		report.Process:        processNodeSummary,
 		report.Container:      containerNodeSummary,
 		report.ContainerImage: containerImageNodeSummary,
 		report.Pod:            podNodeSummary,
+		report.Service:        serviceNodeSummary,
 		report.Host:           hostNodeSummary,
```

Comment:
https://github.com/weaveworks/scope/pull/1204/files#diff-f104a6024333d20eb4e87d1dfa366ebfR77
and
https://github.com/weaveworks/scope/pull/1204/files#diff-d381aa4876e4f400a35de540793f8977R207

Bit of a hack, as it's not really extensible to grouping by other keys. But I figure we can work that out when we do the dsl.